### PR TITLE
Move common test utils

### PR DESCRIPTION
**Pull request by olpipi**
### Details:
 - move common test utils from src/tests/ie_test_utils to src/test/test_utils
 - rename target from commonTestUtils to common_test_utils
 - split sources to src and headers
 - move common util tests from src/tests/ie_test_utils/common_test_utils/tests to src/tests/util_tests and rename target

### Tickets:
 - [CVS-111358](https://jira.devtools.intel.com/browse/CVS-111358)

### CODE REVIEW

File: src/plugins/intel_gna/tests/unit/CMakeLists.txt [hunk 1]
```diff
@@ -25,7 +25,7 @@ addIeTargetTest(
             PRIVATE
                 ngraphFunctions
                 gmock
-                commonTestUtils_s
+                common_test_utils_s
```
Comment (iefode) [inline src/plugins/intel_gna/tests/unit/CMakeLists.txt:28]: Please check why we need to link `common_test_utils_s` to here? Do we really need both `common_test_utils_s` and `common_test_utils`?

File: src/tests/functional/plugin/conformance/subgraphs_dumper/tests/matchers/generic_single_op.cpp [hunk 2]
```diff
@@ -5,7 +5,7 @@
 #include "gtest/gtest.h"
 #include "matchers/single_op.hpp"
 #include "ngraph/ops.hpp"
-#include "functional_test_utils/include/functional_test_utils/summary/op_info.hpp"
+#include "functional_test_utils/summary/op_info.hpp"
```
Comment (iefode) [inline src/tests/functional/plugin/conformance/subgraphs_dumper/tests/matchers/generic_single_op.cpp:8]: This target will be removed soon. No need to update

File: src/tests/test_utils/common_test_utils/include/common_test_utils/file_utils.hpp [hunk 3]
```diff
@@ -247,7 +247,7 @@ inline std::vector<std::string> splitStringByDelimiter(std::string paths, const
     return splitPath;
 }
 
-inline std::string getModelFromTestModelZoo(const std::string& relModelPath);
+std::string getModelFromTestModelZoo(const std::string& relModelPath);
```
Comment (iefode) [inline src/tests/test_utils/common_test_utils/include/common_test_utils/file_utils.hpp:250]: Some methods here should be replaced by `ov::util`

File: src/bindings/c/tests/CMakeLists.txt [hunk 4]
```diff
@@ -7,7 +7,7 @@ set(TARGET_NAME "InferenceEngineCAPITests")
 
 add_executable(${TARGET_NAME} ie_c_api_test.cpp test_model_repo.hpp test_model_repo.cpp)
 
-target_link_libraries(${TARGET_NAME} PRIVATE openvino_c commonTestUtils gtest_main)
+target_link_libraries(${TARGET_NAME} PRIVATE openvino_c common_test_utils gtest_main)
```
[the remark this answers is not in the document]
Comment (olpipi) [reply src/bindings/c/tests/CMakeLists.txt:10]: This is different targets common_test_utils is a util library for tests. ov_util_tests is a tests executable to test common_test_utils

File: src/tests/CMakeLists.txt [hunk 5]
```diff
@@ -28,9 +28,15 @@ ov_deprecated_no_errors()
 
 add_subdirectory(ngraph_helpers)
 add_subdirectory(ie_test_utils)
+add_subdirectory(test_utils)
```
[the remark this answers is not in the document]
Comment (olpipi) [reply src/tests/CMakeLists.txt:31]: I moved util_tests to test_utils/common_test_utils/tests

File: src/tests/functional/plugin/conformance/subgraphs_dumper/tests/matchers/generic_single_op.cpp (hunk 2 above)
Comment (olpipi) [reply src/tests/functional/plugin/conformance/subgraphs_dumper/tests/matchers/generic_single_op.cpp:8]: CI fails if I don't update it :)

File: src/tests/test_utils/common_test_utils/include/common_test_utils/file_utils.hpp (hunk 3 above)
Comment (olpipi) [reply src/tests/test_utils/common_test_utils/include/common_test_utils/file_utils.hpp:250]: My next PR changes namespaces

File: src/plugins/intel_gna/tests/unit/CMakeLists.txt (hunk 1 above)
Comment (olpipi) [reply src/plugins/intel_gna/tests/unit/CMakeLists.txt:28]: Yes common_test_utils_s  is required for unitTestUtils

File: src/plugins/auto_batch/tests/unit/CMakeLists.txt [hunk 6]
```diff
@@ -4,6 +4,10 @@
 
 set(TARGET_NAME ov_auto_batch_unit_tests)
 
+if(SUGGEST_OVERRIDE_SUPPORTED)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-suggest-override")
+endif()
```
Comment (ilyachur) [inline src/plugins/auto_batch/tests/unit/CMakeLists.txt:9]: Why did you add this flag to auto batch unit tests?

**Review by ilyachur — COMMENTED** [review]
In general LGTM.
@iefode Please take a look

File: src/plugins/auto_batch/tests/unit/CMakeLists.txt (hunk 6 above)
Comment (olpipi) [reply src/plugins/auto_batch/tests/unit/CMakeLists.txt:9]: I faced with this type of errors when I was making this changes. Now I checked and this flag is redundant.

**Review by iefode — APPROVED** [review]
LGTM. Waiting for next PRs
